src/toil/provisioners/gceProvisioner.py

```diff
@@ -0,0 +1,915 @@
+                                           location))
+        if ex_image_family:
+            image = driver.ex_get_image_from_family(ex_image_family)
+        if image and not hasattr(image, 'name'):
+            image = driver.ex_get_image(image)
+        if not hasattr(ex_disk_type, 'name'):
+            ex_disk_type = driver.ex_get_disktype(ex_disk_type, zone=location)
+
+        node_attrs = {'size': size,
+                      'image': image,
+                      'location': location,
+                      'network': ex_network,
+                      'subnetwork': ex_subnetwork,
+                      'tags': ex_tags,
+                      'metadata': ex_metadata,
+                      'ignore_errors': ignore_errors,
+                      'use_existing_disk': use_existing_disk,
+                      'external_ip': external_ip,
+                      'ex_disk_type': ex_disk_type,
+                      'ex_disk_auto_delete': ex_disk_auto_delete,
+                      'ex_service_accounts': ex_service_accounts,
+                      'description': description,
+                      'ex_can_ip_forward': ex_can_ip_forward,
+                      'ex_disks_gce_struct': ex_disks_gce_struct,
+                      'ex_nic_gce_struct': ex_nic_gce_struct,
+                      'ex_on_host_maintenance': ex_on_host_maintenance,
+                      'ex_automatic_restart': ex_automatic_restart,
+                      'ex_preemptible': ex_preemptible}
+        # List for holding the status information for disk/node creation.
+        status_list = []
+
+        for i in range(number):
+            name = 'w' + bytes(uuid.uuid4()) #'%s-%03d' % (base_name, i)
+            status = {'name': name, 'node_response': None, 'node': None}
+            status_list.append(status)
+
+        start_time = time.time()
+        complete = False
+        while not complete:
+            if (time.time() - start_time >= timeout):
+                raise Exception("Timeout (%s sec) while waiting for multiple "
+                                "instances")
+            complete = True
+            time.sleep(poll_interval)
+            for status in status_list:
+                # Create the node or check status if already in progress.
+                if not status['node']:
+                    if not status['node_response']:
+                        driver._multi_create_node(status, node_attrs)
+                    else:
+                        driver._multi_check_node(status, node_attrs)
+                # If any of the nodes have not been created (or failed) we are
+                # not done yet.
+                if not status['node']:
+                    complete = False
+
+        # Return list of nodes
+        node_list = []
+        for status in status_list:
+            node_list.append(status['node'])
+        return node_list
+
+
+    ## UNCHANGED CLASSES
+
+    @classmethod
+    def _waitForSSHKeys(cls, instanceIP, keyName='core'):
```

Comment:
This timeout should be added to aws.  Another advantage of making this a part of the abstract provisioner.